lib/collection/src/collection/shard_transfer.rs

```diff
@@ -57,27 +57,28 @@ impl Collection {
             shard_transfer.method.replace(method);
         }
 
-        let shard_id = shard_transfer.shard_id;
         let do_transfer = {
+            let this_peer_id = consensus.this_peer_id();
+            let is_receiver = this_peer_id == shard_transfer.to;
+            let is_sender = this_peer_id == shard_transfer.from;
+
+            // Get the source and target shards, in case of resharding the target shard is different
+            let from_shard_id = shard_transfer.shard_id;
+            let to_shard_id = shard_transfer
+                .to_shard_id
+                .unwrap_or(shard_transfer.shard_id);
+
             let shards_holder = self.shards_holder.read().await;
+            let from_replica_set = shards_holder.get_shard(&from_shard_id).ok_or_else(|| {
+                CollectionError::service_error(format!("Shard {from_shard_id} doesn't exist"))
+            })?;
+            let to_replica_set = shards_holder.get_shard(&to_shard_id).ok_or_else(|| {
+                CollectionError::service_error(format!("Shard {to_shard_id} doesn't exist"))
+            })?;
```

Comment:
Because we now have transfers with a different source and target shard ID (in case resharding), we need a separate source and target replica set in our shard transfer logic. This theme will repeat throughout this PR in various places.